indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -78,164 +85,354 @@ public WorkerTaskMonitor(
     this.cf = cf;
     this.workerCuratorCoordinator = workerCuratorCoordinator;
     this.taskRunner = taskRunner;
-
-    this.exec = Execs.multiThreaded(workerConfig.getCapacity(), "WorkerTaskMonitor-%d");
+    this.exec = Execs.singleThreaded("WorkerTaskMonitor");
   }
 
   /**
    * Register a monitor for new tasks. When new tasks appear, the worker node announces a status to indicate it has
-   * started the task. When the task is complete, the worker node updates the status. It is up to the coordinator to
-   * determine how many tasks are sent to each worker node and cleanup tasks and statuses in ZK accordingly.
+   * started the task. When the task is complete, the worker node updates the status.
    */
   @LifecycleStart
   public void start()
   {
-    try {
-      // restore restorable tasks
-      final List<Pair<Task, ListenableFuture<TaskStatus>>> restored = taskRunner.restore();
-      for (Pair<Task, ListenableFuture<TaskStatus>> pair : restored) {
-        submitTaskRunnable(pair.lhs, pair.rhs);
-      }
+    synchronized (this) {
+      Preconditions.checkState(!started, "already started");
```

Comment:
Suggest checking if exec was shutdown. Otherwise start->stop->start is a "valid" sequence